similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -204,4 +237,24 @@ private long count(SolrQuery query)
         }
         return 0;
     }
+
+    private void appendGenesToQuery(PatientData<Map<String, String>> allGenes, StringBuilder q)
+    {
+        if (allGenes != null && allGenes.size() > 0 && allGenes.isIndexed()) {
+            for (Map<String, String> gene : allGenes) {
+                String geneName = gene.get("gene");
+                String status = gene.get("status");
+                if ("solved".equals(status) || "candidate".equals(status)) {
+                    q.append(" OR solved_genes:"
+                        + ClientUtils.escapeQueryChars(geneName));
+                    q.append(" OR candidate_genes:"
+                        + ClientUtils.escapeQueryChars(geneName));
+                }
+            }
+        }
+
+        if (q.indexOf(" OR ") == 1) {
+            q.delete(1, 5);
```

Comment:
Hmm, this seems unnecessarily complex and opaque. The OR operators are not necessary, nor is specify the field names multiple times. I'll send along a suggestion.